mopidy_spotify/translator.py

```diff
@@ -315,11 +322,14 @@ def web_to_album(web_album):
     ]
     artists = [a for a in artists if a]
 
-    return models.Album(uri=ref.uri, name=ref.name, artists=artists)
+    # Note: date can by YYYY-MM-DD, YYYY-MM or YYYY.
+    date = web_album.get("release_date", "").split("-")[0] or None
```

Comment:
Turns out I do have some local tracks in one of my playlists! And they are annoying to handle because all the fields are present, but most are None. 
```
SpotifyBackend-4 ERROR    2020-12-04 10:19:15,616 Unhandled exception in SpotifyBackend (urn:uuid:2a48d2d3-a581-489a-b0ed-1a3ad4d321ba):
Traceback (most recent call last):
  File "/home/nick/.virtualenvs/mopidy3/lib/python3.8/site-packages/pykka/_actor.py", line 186, in _actor_loop
    self.on_start()
  File "/home/nick/Dev/mopidy-dev/mopidy-spotify/mopidy_spotify/backend.py", line 68, in on_start
    self.playlists.refresh()
  File "/home/nick/Dev/mopidy-dev/mopidy-spotify/mopidy_spotify/playlists.py", line 140, in refresh
    self._get_playlist(playlist_ref.uri)
  File "/home/nick/Dev/mopidy-dev/mopidy-spotify/mopidy_spotify/playlists.py", line 49, in _get_playlist
    return playlist_lookup(
  File "/home/nick/Dev/mopidy-dev/mopidy-spotify/mopidy_spotify/playlists.py", line 265, in playlist_lookup
    playlist = translator.to_playlist(
  File "/home/nick/Dev/mopidy-dev/mopidy-spotify/mopidy_spotify/translator.py", line 240, in to_playlist
    tracks = [t for t in tracks if t]
  File "/home/nick/Dev/mopidy-dev/mopidy-spotify/mopidy_spotify/translator.py", line 240, in <listcomp>
    tracks = [t for t in tracks if t]
  File "/home/nick/Dev/mopidy-dev/mopidy-spotify/mopidy_spotify/translator.py", line 233, in <genexpr>
    web_to_track(
  File "/home/nick/Dev/mopidy-dev/mopidy-spotify/mopidy_spotify/translator.py", line 341, in web_to_track
    album = web_to_album(web_track.get("album", {}))
  File "/home/nick/Dev/mopidy-dev/mopidy-spotify/mopidy_spotify/translator.py", line 326, in web_to_album
    date = web_album.get("release_date", "").split("-")[0] or None
AttributeError: 'NoneType' object has no attribute 'split'
```
Caused by the following web_track:
```
{'album': {'album_type': None, 'artists': [], 'available_markets': [], 'external_urls': {}, 'href': None, 'id': None, 'images': [], 'name': 'Never Mind The Bollocks [Heres The Sex Pistols]', 'release_date': None, 'release_date_precision': None, 'type': 'album', 'uri': None}, 'artists': [{'external_urls': {}, 'href': None, 'id': None, 'name': 'Sex Pistols', 'type': 'artist', 'uri': None}], 'available_markets': [], 'disc_number': 0, 'duration_ms': 211000, 'explicit': False, 'external_ids': {}, 'external_urls': {}, 'href': None, 'id': None, 'is_local': True, 'name': 'Anarchy In The U.K.', 'popularity': 0, 'preview_url': None, 'track_number': 0, 'type': 'track', 'uri': 'spotify:local:Sex+Pistols:Never+Mind+The+Bollocks+%5BHeres+The+Sex+Pistols%5D:Anarchy+In+The+U.K.:211'}
```
This issue has always been there but we just never bumped into it. I submitted https://github.com/mopidy/mopidy-spotify/pull/290 to deal with it since it's really a more general bug.